ethcore/snapshot/snapshot-tests/src/helpers.rs

```diff
@@ -181,14 +181,22 @@ pub fn restore(
 	for state_chunk_hash in manifest.state_hashes.iter() {
 		trace!(target: "snapshot", "state chunk hash: {}", state_chunk_hash);
 		let chunk = reader.chunk(*state_chunk_hash).unwrap();
-		let len = snappy::decompress_into(&chunk, &mut snappy_buffer).unwrap();
+		let required_len = snap::raw::decompress_len(&chunk)?;
+		if snappy_buffer.len() < required_len {
```

Comment:
the `if` is redundant because the `Vec::resize_with` does the same

Reply:
Normally it will also truncate if the supplied length is less than the current length of the vector. I wanted the buffer to be growth-only.